src/common/contentPreference.ts

```diff
@@ -0,0 +1,127 @@
+import { AuthContext } from '../Context';
+import {
+  ContentPreferenceStatus,
+  ContentPreferenceType,
+} from '../entity/contentPreference/types';
+import { ContentPreferenceUser } from '../entity/contentPreference/ContentPreferenceUser';
+import { NotificationPreferenceUser } from '../entity/notifications/NotificationPreferenceUser';
+import {
+  NotificationPreferenceStatus,
+  NotificationType,
+} from '../notifications/common';
+
+type FollowEntity = ({
+  ctx,
+  id,
+  status,
+}: {
+  ctx: AuthContext;
+  id: string;
+  status: ContentPreferenceStatus.Follow | ContentPreferenceStatus.Subscribed;
+}) => Promise<void>;
+
+type UnFollowEntity = ({
+  ctx,
+  id,
+}: {
+  ctx: AuthContext;
+  id: string;
+}) => Promise<void>;
+
+const entityToNotificationTypeMap: Record<
+  ContentPreferenceType,
+  NotificationType[]
+> = {
+  [ContentPreferenceType.User]: [NotificationType.UserPostAdded],
+};
+
+const followUser: FollowEntity = async ({ ctx, id, status }) => {
+  await ctx.con.transaction(async (entityManager) => {
+    const repository = entityManager.getRepository(ContentPreferenceUser);
+
+    const contentPreference = repository.create({
+      userId: ctx.userId,
+      referenceId: id,
+      referenceUserId: id,
+      status,
+    });
+
+    await repository.save(contentPreference);
+
+    if (status === ContentPreferenceStatus.Subscribed) {
+      const notificationRepository = entityManager.getRepository(
+        NotificationPreferenceUser,
+      );
+
+      const notificationPreferences = entityToNotificationTypeMap.user.map(
+        (notificationType) => {
+          return notificationRepository.create({
+            userId: ctx.userId,
+            referenceId: id,
+            notificationType,
+            status: NotificationPreferenceStatus.Subscribed,
+          });
+        },
+      );
+
+      await notificationRepository.save(notificationPreferences);
```

Comment:
This question still relevant?

Reply:
yes cc: @rebelchris @nensidosari

Reply:
If ContentPreference row has status `Subscribed` we automatically then allow all types of follow notifications.

Currently we only notify  user followers (subscribes) when user adds a new post.

Let’s say in a future we also want to notify when he comments. 

User X already follows and is subscribed to user Y. But he only has NotificationPreference entry for user_post_added type. So in case of adding NotificationPreference rows with `Subscribed` we would have to backfill. 

With above idea we treat absence of NotificationPreference row in case of ContentPreference as user is subscribed to everything.

We still allow adding `NotificationPreference.Muted` which would block specific notification.

So its just a matter of optimization of storage and amount of rows we have for each user when they follow different entities.

Reply:
Mentioned also in [RFC](https://dailydotdev.atlassian.net/wiki/spaces/HAN/pages/1126432769/RFC-067+-+Follow?focusedCommentId=1145536546)

Reply:
Yeah sounds good to me as well